debian/cloud-init.postinst

```diff
@@ -7,54 +7,12 @@ set -e
 set -f # disable pathname expansion
 db_capb escape # to support carriage return / multi-line values
 
-debug() {
-   [ "${_CI_UPGRADE_DEBUG:-0}" = "0" ] && return 0
-   echo "$@" 1>&2 || :
-}
-
-update_cfg() {
-   # takes filename, header, new object (in yaml), optionally 'remover'
-   # and merges new into existing object in filename, and then updates file
-   # remover a string that means "delete existing entry"
-   python3 -c '
-import sys, yaml
-
-def update(src, cand):
-    if not (isinstance(src, dict) and isinstance(cand, dict)):
-        return cand
-    for k, v in cand.items():
-        # if the candidate has _ as value, delete source
-        if v == REMOVER:
-            if k in src:
-                del src[k]
-            continue
-        if k not in src:
-            src[k] = v
-        else:
-            src[k] = update(src[k], v)
-    return src
-
-(fname, header, newyaml) = sys.argv[1:4]
-REMOVER = object
-if len(sys.argv) == 5:
-   REMOVER = sys.argv[4]
-newcfg = yaml.safe_load(newyaml)
-
-with open(fname, "r") as fp:
-    cfg = yaml.safe_load(fp)
-if not cfg: cfg = {}
-
-cfg = update(cfg, newcfg)
-
-with open(fname, "w") as fp:
-    fp.write(header + "\n")
-    fp.write(yaml.dump(cfg))' "$@"
-}
-
 fix_1336855() {
   ### Begin fix for LP: 1336855
-  # fix issue where cloud-init misidentifies the location of grub and
+  # fix issue where cloud-init misidentified the location of grub and
```

Comment:
This isn't relevant anymore for images initialized using cloud_style_initialization, and I'm not convinced that we want it on those that aren't, either.

Reply:
Hmm, true. My approach was unnecessarily cautious.